service/controller/resource/statuscondition/create_test.go

```diff
@@ -715,15 +703,57 @@ func TestComputeVersionChange(t *testing.T) {
 					unittest.GetCreatingCondition(90),
 				},
 				Versions: []infrastructurev1alpha2.CommonClusterStatusVersion{
+					unittest.GetVersion(10, "8.7.6"),
 					unittest.GetVersion(60, "8.7.5"),
-					unittest.GetVersion(60, "8.7.6"),
 				},
 			},
 			nodesReady:     true,
 			desiredVersion: "8.7.6",
 
 			expectedResult: false,
 		},
+		// the cluster has not transitioned yet
+		{
+			name: "case 4",
+
+			status: infrastructurev1alpha2.CommonClusterStatus{
+				Conditions: []infrastructurev1alpha2.CommonClusterStatusCondition{
+					unittest.GetUpdatingCondition(30),
+					unittest.GetCreatedCondition(60),
+					unittest.GetCreatingCondition(90),
+				},
+				Versions: []infrastructurev1alpha2.CommonClusterStatusVersion{
+					unittest.GetVersion(60, "8.7.5"),
+				},
+			},
+			nodesReady:     true,
+			desiredVersion: "8.7.6",
+
+			expectedResult: false,
+		},
+		// we roll back
+		{
+			name: "case 5",
+
+			status: infrastructurev1alpha2.CommonClusterStatus{
+				Conditions: []infrastructurev1alpha2.CommonClusterStatusCondition{
+					unittest.GetUpdatedCondition(5),
+					unittest.GetUpdatingCondition(15),
+					unittest.GetUpdatedCondition(20),
+					unittest.GetUpdatingCondition(40),
+					unittest.GetCreatedCondition(60),
+					unittest.GetCreatingCondition(90),
+				},
+				Versions: []infrastructurev1alpha2.CommonClusterStatusVersion{
+					unittest.GetVersion(20, "8.7.6"),
+					unittest.GetVersion(60, "8.7.5"),
+				},
```

Comment:
Does this mean the scenario produces this version history? 

- 8.7.5
- 8.7.6
- 8.7.5

Reply:
I expected that, as adding the new version is triggered here. However, it currently doesn't happen because in `apiextensions` the function `withVersion()` will only add the version to the history if it isn't already there. I am thinking to change that so we can reflect rollbacks in the history. WDYT?

Reply:
This should then be patched in `apiextensions` for sure. 👍

Reply:
Yes, agreed